src/sdk/sdk.test.ts

```diff
@@ -179,7 +180,57 @@ describe("getOwnerRefFrom", () => {
       },
     ]);
   });
+
+  const crWithoutOptionals = {
+    apiVersion: "v1",
+    kind: "Package",
+    metadata: { name: "test", namespace: "default", uid: "1" },
+  };
+
+  const crWithController = { ...crWithoutOptionals, controller: true };
+  const crWithBlockOwnerDeletion = { ...crWithoutOptionals, blockOwnerDeletion: false };
+  const crWithAllFields = { ...crWithoutOptionals, controller: true, blockOwnerDeletion: true };
+
+  const ownerRefWithoutOptionals = [
+    {
+      apiVersion: "v1",
+      kind: "Package",
+      name: "test",
+      uid: "1",
+    },
+  ];
+
+  const ownerRefWithController = ownerRefWithoutOptionals.map(item => ({
+    ...item,
+    controller: true,
+  }));
+  const ownerRefWithBlockOwnerDeletion = ownerRefWithoutOptionals.map(item => ({
+    ...item,
+    blockOwnerDeletion: false,
+  }));
+  const ownerRefWithAllFields = ownerRefWithoutOptionals.map(item => ({
+    ...item,
+    controller: true,
+    blockOwnerDeletion: true,
+  }));
```

Comment:
Those whole block feels tedious, but not necessarily better than when the tests weren't parameterized. For reference, [here's](https://github.com/defenseunicorns/pepr/pull/1104/commits/86ff9632998df4bc13f5d19d8a0c34e136d43449) what that looked like.

Reply:
Discussed with @btlghrants for ways to make this block more concise.